File: it.unibo.bls19Local/src/main/java/it/unibo/blsFramework/listener/ButtonObserver.java
package it.unibo.blsFramework.listener;

import it.unibo.blsFramework.interfaces.IAppLogic;
import it.unibo.blsFramework.interfaces.IApplListener;
import java.util.Observable;

public class ButtonObserver implements IApplListener {
private IAppLogic controller;
	private int count   = 0;
//Factory method
	public static ButtonObserver createButtonListener(  ){
		return new ButtonObserver(   );
	}
	private ButtonObserver(  ){
		super();
 	}

	public void setControl( IAppLogic ctrl ){
		this.controller = ctrl;
	}

	public int getNumOfClicks(){
		return count;
	}

	@Override  //from IButtonListener IObserver -> Observer
	public void update( Observable source, Object state ){
		count++;
		//System.out.println("ButtonListener update | state=" + state + " controller=" + controller);
		controller.execute ( state.toString() );
	}
}

File: it.unibo.bls19Local/src/main/kotlin/it/unibo/bls/kotlin/applLogic/ButtonObserver.kt
package it.unibo.bls.kotlin.applLogic
import it.unibo.blsFramework.interfaces.IAppLogic
import java.util.Observable

class ButtonObserver private constructor() : IApplListener {
    private var controller: IAppLogic? = null
    private var count = 0

    override fun setControl(ctrl: IAppLogic) {
        this.controller = ctrl
    }

    override fun getNumOfClicks(): Int {
        return count
    }

    override//from IApplListener IObserver -> Observer
    fun update(source: Observable, state: Any) {
        count++
        //System.out.println("ButtonListener update | state=" + state + " controller=" + controller);
        controller!!.execute(state.toString())
    }

    companion object {
        //Factory method
        fun createButtonListener(): IApplListener {
            return ButtonObserver()
        }
    }
}
